demos/python_demos/object_detection_retinaface_demo/postprocessor.py

```diff
@@ -0,0 +1,253 @@
+"""
+
+import numpy as np
+import re
+
+class FacialLandmarksPostprocessor(object):
+    def __init__(self, detect_masks=False):
+        self._detect_masks = detect_masks
+        _ratio = (1.,)
+        self._anchor_cfg = {
+            32: {'SCALES': (32, 16), 'BASE_SIZE': 16, 'RATIOS': _ratio},
+            16: {'SCALES': (8, 4), 'BASE_SIZE': 16, 'RATIOS': _ratio},
+            8: {'SCALES': (2, 1), 'BASE_SIZE': 16, 'RATIOS': _ratio}
+        }
+        self._features_stride_fpn = [32, 16, 8]
+        self._anchors_fpn = dict(zip(self._features_stride_fpn, self.generate_anchors_fpn(cfg=self._anchor_cfg)))
+        self._num_anchors = dict(zip(
+            self._features_stride_fpn, [anchors.shape[0] for anchors in self._anchors_fpn.values()]
+        ))
+        self.landmark_std = 0.2 if detect_masks else 1.0
+
+    @staticmethod
+    def generate_anchors_fpn(cfg):
+        def generate_anchors(base_size=16, ratios=(0.5, 1, 2), scales=2 ** np.arange(3, 6)):
+            base_anchor = np.array([1, 1, base_size, base_size]) - 1
+            ratio_anchors = _ratio_enum(base_anchor, ratios)
+            anchors = np.vstack([_scale_enum(ratio_anchors[i, :], scales) for i in range(ratio_anchors.shape[0])])
+            return anchors
+
+        def _ratio_enum(anchor, ratios):
+            w, h, x_ctr, y_ctr = _generate_wh_ctrs(anchor)
+            size = w * h
+            size_ratios = size / ratios
+            ws = np.round(np.sqrt(size_ratios))
+            hs = np.round(ws * ratios)
+            anchors = _make_anchors(ws, hs, x_ctr, y_ctr)
+            return anchors
+
+        def _scale_enum(anchor, scales):
+            w, h, x_ctr, y_ctr = _generate_wh_ctrs(anchor)
+            ws = w * scales
+            hs = h * scales
+            anchors = _make_anchors(ws, hs, x_ctr, y_ctr)
+            return anchors
+
+        def _generate_wh_ctrs(anchor):
+            w = anchor[2] - anchor[0] + 1
+            h = anchor[3] - anchor[1] + 1
+            x_ctr = anchor[0] + 0.5 * (w - 1)
+            y_ctr = anchor[1] + 0.5 * (h - 1)
+            return w, h, x_ctr, y_ctr
+
+        def _make_anchors(ws, hs, x_ctr, y_ctr):
+            ws = ws[:, np.newaxis]
+            hs = hs[:, np.newaxis]
+            anchors = np.hstack((
+                x_ctr - 0.5 * (ws - 1), y_ctr - 0.5 * (hs - 1), x_ctr + 0.5 * (ws - 1), y_ctr + 0.5 * (hs - 1)
+            ))
+            return anchors
+
+        rpn_feat_stride = [int(k) for k in cfg]
+        rpn_feat_stride.sort(reverse=True)
+        anchors = []
+        for stride in rpn_feat_stride:
+            feature_info = cfg[stride]
+            bs = feature_info['BASE_SIZE']
+            __ratios = np.array(feature_info['RATIOS'])
+            __scales = np.array(feature_info['SCALES'])
+            anchors.append(generate_anchors(bs, __ratios, __scales))
+
+        return anchors
+
+    @staticmethod
+    def nms(x1, y1, x2, y2, scores, thresh, include_boundaries=True):
```

Comment:
since this method used only here and `include_boundaries` = False in call,  this parameter can be removed